src/components/checkbox-group/CheckboxGroup.tsx

```diff
@@ -0,0 +1,282 @@
+import {
+  type ValidationProps,
+  FormField,
+  FormFieldValidation,
+  FormFieldWidth,
+} from '../form/FormField';
+import { useFormFieldA11y } from '../form/useFormFieldA11y';
+
+/**
+ * A checkbox group allows a user to select multiple items from a list of
+ * options. A checkbox group must have at least one option.
+ *
+ * Each option must have a unique value. If no value is provided for an option,
+ * then the option's label will be used as its value.
+ *
+ * The selected values may appear in any order. Equality is determined using
+ * strict equality (`===`).
+ *
+ * The disabled state and error state will be applied to the entire group.
+ * However, the disabled state or error state can be overridden per option.
+ *
+ * > A checkbox group is a controlled element, it requires a `value` and
+ * `onChange` handler.
+ *
+ * **Permitted Content**
+ *
+ * None; it's a self closing element.
+ *
+ * **Permitted Parents**
+ *
+ * Any element that accepts flow content.
+ *
+ * **Example**
+ *
+ * ```tsx
+ * <CheckboxGroup
+ *   label="Favorite colors"
+ *   onChange={setFavoriteColors}
+ *   options={[
+ *     { label: 'Red' },
+ *     { label: 'Green' },
+ *     { label: 'Blue' },
+ *   ]}
+ *   value={favoriteColors}
+ * />
+ * ```
+ */
+export const CheckboxGroup: CheckboxGroup = forwardRef<HTMLDivElement, CheckboxGroupProps>(
+  (props, ref) => {
+    const {
+      ariaLabel,
+      ariaLabelledBy,
+      className,
+      direction,
+      helpText,
+      id,
+      isDisabled,
+      isRequired,
+      label,
+      messageInvalid,
+      messageValid,
+      name,
+      onChange,
+      options,
+      serializeValue = defaultSerializer,
+      validation,
+      value,
+      width,
+    } = props;
+
+    const defaultId = `${useId()}checkbox`;
+
+    const { controlId, describedBy, descriptionId, labelId, validationMessageId } =
+      useFormFieldA11y({
+        controlId: id ?? defaultId,
+        hasDescription: Boolean(helpText),
+        hasValidationMessage: Boolean(messageInvalid || messageValid),
+      });
+
+    const serializedValue = useMemo(() => {
+      return name ? value.map(serializeValue) : [];
+    }, [name, serializeValue, value]);
+
+    return (
+      <FormField
+        className={className}
+        description={helpText}
+        descriptionId={descriptionId}
+        isDisabled={isDisabled}
+        isRequired={isRequired}
+        label={label}
+        labelId={labelId}
+        messageInvalid={messageInvalid}
+        messageValid={messageValid}
+        validation={validation}
+        validationMessageId={validationMessageId}
+        width={width}>
+        <CheckboxGroupRoot // NOSONAR
+          aria-describedby={describedBy}
+          aria-label={ariaLabel}
+          aria-labelledby={ariaLabelledBy ?? labelId}
+          data-direction={direction}
+          data-invalid={validation === FormFieldValidation.Invalid}
+          id={controlId}
+          ref={ref}
+          role="group">
```

Comment:
I have a `// NOSONAR` here because it is complaining about the `group` role.

Reply:
Let's remove it, we prefer managing these directly in SQ UI by "accepting" the issue and eventually explaining why, instead of in the code.

Reply:
I did this but SonarQube still complains about it in IDE

![image](https://github.com/user-attachments/assets/2e8d6633-a652-4975-97bc-608064f10892)

Reply:
Mmmh if SonarQube keeps complaining in IDE after accepting the issue in the UI it's probably because your Sonarlint isn't in connected mode. Lets try to fix that instead of using nosonar